### ISSUE TITLE
[HOLD for payment 2023-09-29] [Image] Feature Request: Introduce an ability to view workspace picture with a single tap - reported by @chauchausoup

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#11197]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem: 
We cannot see a workspace avatar picture in tap and view manner. Once a user uploads a workspace avatar image then there is no way for them to view it by any means.

## Solution: 
As in profile picture, avatar can be viewed by tapping on it while being on chat. We should implement similar feature with workspace avatar image too. We can make view ability if a user taps in Avatar under general settings. Only if they taps in camera icon then the usual Upload / Remove option should be shown.

##Context: 
[https://expensify.slack.com/files/U040F2ZTA3G/F040ZBKE2CB/rpreplay_final1662273851.mp4](https://expensify.slack.com/files/U040F2ZTA3G/F040ZBKE2CB/rpreplay_final1662273851.mp4?origin_team=T03SC9DTT&origin_channel=C01GTK53T8Q)

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation  

https://user-images.githubusercontent.com/43995119/191662086-9952bd00-57a9-4af5-a85c-86758e95a232.MP4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @chauchausoup
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1662273908091079

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Gonals** [comment]
Setting as external!

**Comment by Santhosh-Sellavel** [comment]
@NicMendonca 

Busy with existing work items, also limited availability for next couple weeks So un-assigning this one. Can you reassign this to another C+ by reapplying the `external` label?

Also, Leave a note there is a proposal yet to be reviewed https://github.com/Expensify/App/issues/11197#issuecomment-1255435384.

Thanks!

**Comment by NicMendonca** [comment]
Job posting: https://www.upwork.com/jobs/~01960cad105fe6709e

**Comment by Puneet-here** [comment]
> "Workspace avatar" is good.

This functionality will be present for profile page too

**Comment by Puneet-here** [comment]
Nothing. We can do that as well. We just have to pass a prop.

**Comment by stitesExpensify** [comment]
Not overdue, waiting for c+ review

**Comment by stitesExpensify** [comment]
Sounds good, thanks @rushatgabhane. @shawnborton does this seem like a good solution to you?

**Comment by shawnborton** [comment]
That makes sense to me, let's do it.

**Comment by stitesExpensify** [comment]
Sweet! @Puneet-here go ahead and apply on Upwork!

**Comment by Beamanator** [comment]
Has any work been done here yet? If not, I think we should put this on HOLD since we're handing lots of image-related projects in this tracking issue: [Image-related feature request and improvements](https://github.com/Expensify/App/issues/10894#top)

**Comment by NicMendonca** [comment]
@Puneet-here based off @Beamanator's comment - have you got started on this one?

**Comment by NicMendonca** [comment]
@Puneet-here any update?

**Comment by Puneet-here** [comment]
@NicMendonca aren't we holding this? Should I push the code?

**Comment by mallenexpensify** [comment]
@NicMendonca Snaggin' from you cuz I got the tracking issue assigned to me
[Tracking] [Polish] Image-related feature request and improvements #10894

@Puneet-here , hold off on pushing code for now. Thanks for the patience

**Comment by mallenexpensify** [comment]
on hold

**Comment by trjExpensify** [comment]
On hold, Melv.

**Comment by trjExpensify** [comment]
Still held.

**Comment by trjExpensify** [comment]
Feature request still held on completing the more pressing bugs tied to the image improvements initiative.

**Comment by stitesExpensify** [comment]
On hold

**Comment by trjExpensify** [comment]
Feature request that's still on hold, Melv.

**Comment by trjExpensify** [comment]
Still held.

**Comment by trjExpensify** [comment]
Still held.

**Comment by stitesExpensify** [comment]
Hold

**Comment by trjExpensify** [comment]
Same, Melv.

**Comment by trjExpensify** [comment]
Same, Melv.

**Comment by trjExpensify** [comment]
So @Beamanator now that the image related improvements is down to caching, and held on the server migration to complete that. What do you think about opening this back up, as it's standalone?

**Comment by Beamanator** [comment]
That's fair to me @trjExpensify 👍

**Comment by stitesExpensify** [comment]
@Puneet-here do you think that your proposal is still accurate after all the changes that have happened between then and now?

**Comment by Puneet-here** [comment]
Yes, it still works.

**Comment by Puneet-here** [comment]
Contributor details
Your Expensify account email: puneet.here121@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~013a9f7d8a0f7cbcb7

**Comment by trjExpensify** [comment]
Okay, so are we going to proceed with @Puneet-here's proposal? @rushatgabhane @stitesExpensify?

**Comment by trjExpensify** [comment]
@rushatgabhane @stitesExpensify bump, can @Puneet-here proceed with his proposal?

**Comment by trjExpensify** [comment]
Cool, thanks! @Puneet-here crack on with the PR, please! :)

**Comment by Puneet-here** [comment]
Hi, sorry for the delay here. I was having some problems with my system but I have resolved it and I will be able to raise the PR tonight.

**Comment by trjExpensify** [comment]
Thanks for keeping us in the loop, @Puneet-here. Did you create that PR?

**Comment by Puneet-here** [comment]
Everything is set but there is one weird problem which I am trying to investigate. So at the profile page when I upload the image and try to view it just after the upload, I don't see the image in the preview it just shows attachment icon but when I pass the file name of the profile page avatar at `AttachmentModal` in `originalFileName` prop, then it works for profile page and workspace page both

**Comment by Puneet-here** [comment]
Hi sorry for delaying this too much. I am not getting enough time because of my sister's wedding. I will be able to raise the PR in 2 days. I am trying my best. If it's too late then I would be happy to share the code and some other contributor can make the last changes.

**Comment by Puneet-here** [comment]
Yes. I am making the PR on a different machine so it is taking me some time setting up the things. It will be done within one hour.

**Comment by Puneet-here** [comment]
PR is up.

**Comment by trjExpensify** [comment]
Where are we at with the PR review?

**Comment by stitesExpensify** [comment]
Looks like @rushatgabhane started their review today. For some reason I was not auto-assigned as a reviewer, so I added myself and will test once Rushat is done.

**Comment by trjExpensify** [comment]
Commented on the PR, seems like we're trying to repro a bug @rushatgabhane is seeing but @Puneet-here is not. :)

**Comment by trjExpensify** [comment]
@rushatgabhane this one has been lingering in review for a bit, how you getting on?

**Comment by rushatgabhane** [comment]
yeah this one was dragged quite a bit 😅
will get it done today

**Comment by trjExpensify** [comment]
@Puneet-here looks like we have more conflicts! :)

**Comment by trjExpensify** [comment]
@Puneet-here @rushatgabhane mind giving us an update on progress from the linked PR? Has it stalled again, what are the next steps? Thanks!

**Comment by trjExpensify** [comment]
I kinda' think the primary action here is to `Upload photo` if you're clicking the avatar versus simply viewing it. I wouldn't be against putting it second in the list, but not super passionate. @shawnborton, do you have an opinion on that?

**Comment by shawnborton** [comment]
Second in the list works for me, but I also don't feel strongly.

**Comment by trjExpensify** [comment]
PR is being held for the merge freeze, but we should wrap-up here in a few days melv!

**Comment by trjExpensify** [comment]
@Puneet-here there are conflicts to resolve, please jump on them and let's get this finished. Thanks!

**Comment by Puneet-here** [comment]
Updated! Sorry for the delay from my side, I am out right now.

**Comment by trjExpensify** [comment]
Alright, seems like we're ready to make payments here! Old bounty levels of $1,000 apply on this issue because it was created prior to August 30th, but it's still a `new feature` request which we don't pay for reporting as bugs. 

This PR [took longer than 9 days](https://github.com/Expensify/App/issues/11197#issuecomment-1724800271), so a 50% reduction in the bounty applies per [here](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#payment-for-contributions). That makes the payouts by my calculation: 

- @Puneet-here - $500 for the implementation
- @rushatgabhane - $500 for the review

Everyone agree on that? Let me know, and I'll send @Puneet-here an offer in Upwork, and Rushat you can proceed to request.

**Comment by chauchausoup** [comment]
@trjExpensify while I was reporting the Feature Request , it was written in FR guideline that FR is viable for a bounty.

**Comment by trjExpensify** [comment]
Oh hm, fair enough. That policy went into place in Nov 2022 [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1668126086760699). This issue was created in Sept 22 originally, so yes, $250 is due for the report. 

Added that to my post above and sent you an offer. Waiting for confo from @rushatgabhane on the payments above to proceed!

**Comment by chauchausoup** [comment]
Thank you @trjExpensify .

**Comment by trjExpensify** [comment]
@chauchausoup paid! 
@Puneet-here offer sent. 
@rushatgabhane free to proceed to request.

**Comment by Puneet-here** [comment]
Accepted.

**Comment by trjExpensify** [comment]
Paid!

**Comment by JmillsExpensify** [comment]
$500 payment approved for @rushatgabhane based on BZ summary

**Comment by trjExpensify** [comment]
Okay great, closing now then!
